lib/declarative_authorization/test/helpers.rb

```diff
@@ -155,8 +155,14 @@ def this_is_an_abstract_controller_so_it_needs_no_access_tests
         alias :access_tests_not_required :this_is_an_abstract_controller_so_it_needs_no_access_tests
 
         def all_public_actions
-          actions = controller_class.public_instance_methods(false)
-          actions += controller_class.superclass.public_instance_methods(false)
+          actions = []
+          if defined?(Grape) && [Grape::API, Grape::API::Instance].any? { |base| controller_class < base }
+            actions += controller_class.routes.map { |api| "#{api.request_method} #{api.origin}" }
```

Comment:
I don't really understand how this works but I guess it adds a string that is somehow parsed somewhere else?

Reply:
In rails controllers, actions are uniquely identified by their method name (eg `index`, `show`, etc). In Grape APIs the endpoints are not generated in the same way, so we instead need to use a combination of the HTTP Verb and Path.

This means authorization rules for APIs look like this:

```ruby
role :tenant do
  allowed to: ['GET /occupancies/:id'], when: [:read, :write_without_delete, :write]
end
```

The string `GET /occupancies/:id` is what is being checked here.

Reply:
Oh right, that string. Yeah makes sense.